airflow/models/xcom.py

```diff
@@ -435,7 +439,7 @@ def get_many(
         include_prior_dates: bool = False,
         limit: int | None = None,
         session: Session = NEW_SESSION,
-    ) -> Query:
+    ) -> Select:
```

Comment:
Hmm, I’m not sure we can change the `get_many` return type like this since it is considered a public interface. We need to make the `session.scalars` call a part of this function instead.

Reply:
Yes, that's why I converted it to draft: https://github.com/apache/airflow/pull/37761#issuecomment-1967921798